src/utils/fileUpload.js

```diff
@@ -65,11 +76,19 @@ const findUniquePath = async function(client, userRoot, path) {
  * @param {object} files the files to be processed
  * @param {string} token the conversation's token where to share the files
  * @param {number} uploadId a unique id for the upload operation indexing
+ * @param {bool} rename whether to rename the files (usually after pasting)
  */
-const processFiles = async function(files, token, uploadId) {
+const processFiles = async function(files, token, uploadId, rename) {
+	if (rename) {
+		files.forEach(file => {
+			// note: can't overwrite the original read-only name attribute
+			file.newName = moment(file.lastModified || file.lastModifiedDate).format('YYYYMMDD_HHmmss')
```

Comment:
@jancborchardt file name pattern okay? or should we do something alike `YYYY-MM-DD HH-mm-ss` ?